packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -44,16 +45,28 @@ export class CookieStorage extends BaseStorage {
         super(options)
 
         if (typeof document === 'undefined') {
-            throw new Error('CookieStorage is not avaliable on the current environment.')
+            throw new Error('CookieStorage is not available on the current environment.')
         }
     }
     set(key: string, value: string, options?: Cookies.CookieAttributes) {
         const suffixedKey = this.getSuffixedKey(key)
+        const parentUrl = document.location?.ancestorOrigins?.[0] || document.referrer
+        const parentHostName = parentUrl ? new URL(parentUrl).hostname : ''
+        const isInAllowList = IFRAME_HOST_ALLOW_LIST.includes(parentHostName)
+
+        const isLocalHost = window.location.hostname === 'localhost'
         Cookies.set(suffixedKey, value, {
             // Deployed sites will always be HTTPS, but the local dev server is served over HTTP.
             // Ideally, this would be `secure: true`, because Chrome and Firefox both treat
             // localhost as a Secure context. But Safari doesn't, so here we are.
             secure: !onClient() || window.location.protocol === 'https:',
+            // By default, Chromes does not allow cookies to be sent/read
+            // when the code is loaded in iframe (e.g storefront preview case)
+            // setting sameSite to none lose that restriction to
+            // make sure that cookies can be read/sent when code is loaded in an iframe of a certain allow host list.
+            // Outside of iframe, we want to keep most browser default value (Chrome or Firefox uses Lax)
+            // https://web.dev/samesite-cookie-recipes/
+            sameSite: !isLocalHost && isInAllowList ? 'none' : 'Lax',
```

Comment:
Lax is more relaxed than "strict" in terms of cookies setting. By defaults, most browsers uses Lax (when sameSite is not set). I think we should set it as such since using Strict may restrict lots of behavior when it comes to cookies

> The SameSite=Strict value will only allow first party cookies to be sent. This setting is good for user actions like login credentials, but the cookie will not be sent on the initial request to the webpage. The SameSite=Lax setting will allow the user to maintain a logged in status while arriving from an external link.